frontend/src/components/common/Resource/EditorDialog.tsx

```diff
@@ -152,45 +164,31 @@ export default function EditorDialog(props: EditorDialogProps) {
     );
 
     setCode({ code: value as string, format: code.format });
+    if (error && getObjectsFromCode({ code: value as string, format: originalCode.format })) {
+      setError('');
+    }
 
     if (onEditorChanged) {
       onEditorChanged(value as string);
     }
   }
 
-  function getObjectFromCode(codeInfo: typeof originalCode) {
-    function looksLikeJson(code: string) {
-      const trimmedCode = code.trimRight();
-      const lastChar = !!trimmedCode ? trimmedCode[trimmedCode.length - 1] : '';
-      const firstChar = !!trimmedCode ? trimmedCode[0] : '';
-      if (['{', '['].includes(firstChar) || ['}', ']'].includes(lastChar)) {
-        return true;
-      }
-
-      return false;
-    }
-
+  function getObjectsFromCode(codeInfo: typeof originalCode): {
+    obj: KubeObjectInterface[] | null;
+    format: string;
+    error: Error | null;
+  } {
     const { code, format } = codeInfo;
-    const res: { obj: KubeObjectInterface | null; format: string; error: Error | null } = {
+    const res: { obj: KubeObjectInterface[] | null; format: string; error: Error | null } = {
       obj: null,
       format,
       error: null,
     };
 
-    if (!res.obj) {
-      res.format = 'yaml';
-      try {
-        res.obj = yaml.load(code) as KubeObjectInterface;
-        return res;
-      } catch (e) {
-        res.error = new Error((e as Error).message || t('Invalid YAML'));
-      }
-    }
-
```

Comment:
By doing this, you're switching the order in which we evaluate the code, which is risky as it can create more unforeseen problems.

Reply:
this was removed by mistake, updated this